### DISCUSSION TITLE
Is there a way to only allow one way resizibility? (Desktop)

### DISCUSSION THREAD

**Discussion by EmrecanKaracayir** [microsoft/WindowsAppSDK#2101] [General] [answered]
AppWindow has "IsResizible" method but it disables resizing all together.

MainWindow has "SizeChanged" event but it reacts AFTER the resizing happens. So I can not interfere with how the resizing applies.

Is there a way to only allow vertical/horizontal resizing of the MainWindow?

**Comment by castorix** [comment, marked as answer]
You can subclass the main window with **SetWindowSubclass** and handle WM_SIZING or WM_GETMINMAXINFO
(main window handle =  _WinRT.Interop.WindowNative.GetWindowHandle(this)_)

**Reply by EmrecanKaracayir** [reply]
Thank you for your answer. Can you give us code examples?

**Reply by castorix** [reply]
A Test with the 2 methods : 
[Test disable (horizontal) resizing](http://pastebin.fr/101995)

**Reply by styris-ame** [reply]
> A Test with the 2 methods : [Test disable (horizontal) resizing](http://pastebin.fr/101995)

Any chance you could repost the sample? The pastebin link is down

**Reply by castorix** [reply]
// Global variables
            private IntPtr hWndMain = IntPtr.Zero;     
            private Microsoft.UI.Windowing.AppWindow _apw;
 
            // At beginning
            hWndMain = WinRT.Interop.WindowNative.GetWindowHandle(this);
            _apw = this.AppWindow;
 
            SubClassDelegate = new SUBCLASSPROC(WindowSubClass);
            bool bRet = SetWindowSubclass(hWndMain, SubClassDelegate, 0, 0);

        // Window procedure
        private int WindowSubClass(IntPtr hWnd, uint uMsg, IntPtr wParam, IntPtr lParam, IntPtr uIdSubclass, uint dwRefData)
        {
            switch (uMsg)
            {
                //case WM_SIZING:
                //    {
                //        if (wParam == (IntPtr)WMSZ_LEFT || wParam == (IntPtr)WMSZ_BOTTOMLEFT || wParam == (IntPtr)WMSZ_TOPLEFT)
                //        {
                //            RECT rc = (RECT)Marshal.PtrToStructure(lParam, typeof(RECT));                            
                //            rc.left = _apw.Position.X;
                //            Marshal.StructureToPtr(rc, lParam, false);
                //        }
                //        else if (wParam == (IntPtr)WMSZ_RIGHT || wParam == (IntPtr)WMSZ_BOTTOMRIGHT || wParam == (IntPtr)WMSZ_TOPRIGHT)
                //        {
                //            RECT rc = (RECT)Marshal.PtrToStructure(lParam, typeof(RECT));
                //            rc.right = _apw.Position.X + _apw.Size.Width;
                //            Marshal.StructureToPtr(rc, lParam, false);
                //        }
                //        return 0;
                //    }
                //    break;

                case WM_GETMINMAXINFO:
                    {
                        MINMAXINFO mmi = (MINMAXINFO)Marshal.PtrToStructure(lParam, typeof(MINMAXINFO));
                        mmi.ptMaxSize.X = mmi.ptMinTrackSize.X = mmi.ptMaxTrackSize.X = _apw.Size.Width;
                        Marshal.StructureToPtr(mmi, lParam, false);
                        return 0;
                    }
                    break;
            }
            return DefSubclassProc(hWnd, uMsg, wParam, lParam);
        }

        // Declarations
        public delegate int SUBCLASSPROC(IntPtr hWnd, uint uMsg, IntPtr wParam, IntPtr lParam, IntPtr uIdSubclass, uint dwRefData);

        [DllImport("Comctl32.dll", SetLastError = true)]
        public static extern bool SetWindowSubclass(IntPtr hWnd, SUBCLASSPROC pfnSubclass, uint uIdSubclass, uint dwRefData);

        [DllImport("Comctl32.dll", SetLastError = true)]
        public static extern int DefSubclassProc(IntPtr hWnd, uint uMsg, IntPtr wParam, IntPtr lParam);
        private SUBCLASSPROC SubClassDelegate;
        
        public const int WM_GETMINMAXINFO = 0x0024;
        public const int WM_SIZING = 0x0214;

        public const int WMSZ_LEFT = 1;
        public const int WMSZ_RIGHT = 2;
        public const int WMSZ_TOP = 3;
        public const int WMSZ_TOPLEFT = 4;
        public const int WMSZ_TOPRIGHT = 5;
        public const int WMSZ_BOTTOM = 6;
        public const int WMSZ_BOTTOMLEFT = 7;
        public const int WMSZ_BOTTOMRIGHT = 8;

        [StructLayout(LayoutKind.Sequential)]
        public struct MINMAXINFO
        {
            public System.Drawing.Point ptReserved;
            public System.Drawing.Point ptMaxSize;
            public System.Drawing.Point ptMaxPosition;
            public System.Drawing.Point ptMinTrackSize;
            public System.Drawing.Point ptMaxTrackSize;
        }
